src/actions/ui.js

```diff
@@ -162,11 +162,15 @@ export function clearProjectDirectoryRoot() {
 export function setProjectDirectoryRoot(newRoot: string) {
   return ({ dispatch, getState }: ThunkArgs) => {
     const curRoot = getProjectDirectoryRoot(getState());
+
+    // What is this for?  It's causing problems by appending successive root settings
+    /*
```

Comment:
Let's say we have this as root: `/example.com/route`
So, our tree nodes will now be updated with the path which will start from `/route` since it's the root.

And now if you try to set project root again, then this functionality comes. So if the new root is `/route/ex`, so we'll append `ex` with the current root.
Hope it makes things clear. I'd love to see if we can make it simpler.